src/HTTPure/Body.purs

```diff
@@ -1,25 +1,64 @@
 module HTTPure.Body
   ( Body
+  , string
+  , binary
+  , isString
+  , isBinary
+  , fromString
+  , fromBinary
   , read
   , write
+  , size
   ) where
 
 import Prelude
 
 import Data.Either as Either
+import Data.String as String
 import Effect as Effect
 import Effect.Aff as Aff
 import Effect.Ref as Ref
+import Node.Buffer as Buffer
 import Node.Encoding as Encoding
 import Node.HTTP as HTTP
 import Node.Stream as Stream
 
 -- | The `Body` type is just sugar for a `String`, that will be sent or received
 -- | in the HTTP body.
-type Body = String
+data Body
+  = StringBody String
+  | BinaryBody Buffer.Buffer
+
+-- | Construct a `Body` from a string
+string :: String -> Body
+string = StringBody
+
+-- | Construct a `Body` from a `Buffer`
+binary :: Buffer.Buffer -> Body
+binary = BinaryBody
+
+-- | Returns `true` when the `Body` is a string
+isString :: Body -> Boolean
+isString (StringBody _) = true
+isString _ = false
+
+-- | Returns `true` when the `Body` is binary
+isBinary :: Body -> Boolean
+isBinary (BinaryBody _) = true
+isBinary _ = false
+
+-- | A partial function that extracts the value from a string body.
+-- | Passing a binary body will throw an error at runtime.
+fromString :: Partial => Body -> String
+fromString (StringBody s) = s
+
+-- | A partial function that extracts the value from a binary body.
+-- | Passing a string body will throw an error at runtime.
+fromBinary :: Partial => Body -> Buffer.Buffer
+fromBinary (BinaryBody b) = b
```

Comment:
Similar to above, these functions feel like a way around exposing the data constructors.  I'd rather just decompose using purescript native syntax and not have these in the API.